File: src/taeeun/week5/캐시.js
function solution(cacheSize, cities) {
    var answer = 0;
    let cache=[]; //캐시 순서는 참조 순

    cities=cities.map(city=>city.toLowerCase());
    
    if (cacheSize===0) return cities.length*5;
    
    for (const city of cities) {
        //캐시에 존재하는지 확인
        const idx=cache.findIndex((c)=>c===city);
        if (idx!==-1) {
            answer+=1;
            //해당 도시를 가장 앞으로
            cache.splice(idx, 1);
            cache.unshift(city);
        }
        else {
            //캐시에 존재 x
            answer+=5;
            if (cache.length>=cacheSize) cache.pop();
            cache.unshift(city);
        }
    }
    return answer;
}

File: src/chanhyun/week5/캐시.py
from collections import defaultdict

def solution(cacheSize, cities):
    answer = 0
    cache = defaultdict()
    
    if cacheSize == 0:
        return len(cities)*5
    
    for clock, city in enumerate(cities):
        city = city.lower()
        if city in cache:
            answer += 1
            cache[city] = clock
        else:
            answer += 5
            if len(cache) < cacheSize:
                cache[city] = clock
            else:
                # LRU
                cacheList = sorted(cache.items(), key=lambda x: x[1])
                del cache[cacheList[0][0]]
                cache[city] = clock
    return answer